doxygen/contributor_help_pages/require_meta.md

```diff
@@ -1,22 +1,263 @@
+are named `is_{condition}` and the struct contains a `value` that is
+`true` or `false` at compile time. For example, `is_var<T>::value` is
+`true` if and only if the type `T` is `stan::math::var_value`.
+
+We provide `requires<>` type traits based on the boolean
+`is_{condition}` type traits. When types satisfy the condition, the
+`requires<>::value` will evaluate to `void`. When the types do not
+satisfy the condition, `requires<>::value` is an invalid subsitution
+and is not used. (See @ref requires_impl for more details.)
+
+Note: every possible requires<> type trait is not implemented in the
+Stan Math library. If one of the missing requires<> type trait is
+missing, we can implement it and include it. Please see @ref
+requires_dev_guide for more information.
+
+
+#### All requires<> type traits
+
+For any boolean type trait, below is the list of possible requires<>
+type traits.  Any `*` should be thought of as a wildcard where a type
+traits name is put in its place. For example, for `is_var`, we can
+substitute `var` for `*`.
+
+1. `require_*_t`: A template parameter `T` must satisfy the `is_*`
+type trait. This means `require_var_t<stan::math::var>::value` is
+`void`, but `require_var_t<double>::value` is an invalid subsitution.
+
+2. `require_not_*_t`: A template parameter `T` must not satisfy the
+`is_*` type trait.
+
+    *NOTE:* The `not` version of the `requires` template parameters
+should be used sparingly.  Often a `requires` template parameter is
+used to specify what types a function should accept.  Defining a
+function by the types it cannot accept can make understanding what
+goes into a function more difficult and error prone.
+
+3. `require_all_*_t`: All types in the parameter pack of types must
+satisfy the `is_*` type trait.
+
+4. `require_any_*_t`: Any type in the parameter pack of types must
+satisfy the `is_*` type trait.
+
+5. `require_any_not_*_t`: Any type in the parameter pack must not
+satisfy the `is_*` type trait.
 
-`requires` template parameters are [`std::enable_if_t`](https://en.cppreference.com/w/cpp/types/enable_if) aliases such as the following example definition of @ref stan::require_t.
+6. `require_all_not_*_t`: All types in the parameter pack must not
+satisfy the `is_*` type trait.
+
+
+    `std::vector` and `Eigen` types have additional `requires`
+    template parameters to detect if the @ref stan::value_type (the
+    type of the elements of either `std::vector` or the `Eigen` type)
+    or the @ref stan::scalar_type (the underlying scalar type after
+    recursively walking through the container types) satisfy a
+    condition to enable a class or function.
+
+    The container `requires` template parameters have an ending at
+    their signature of `_vt` and `_st` to symbolize whether you want
+    to inspect the @ref stan::value_type or @ref stan::scalar_type.
+
+    In the next requires traits, `is_type` is used to represent any
+    boolean type trait.
```

Comment:
Is there a reason this is tabbed?

Reply:
Yes, although it's a hack. I was trying to keep the numbering sequential. I think the enumeration of requires traits is easier if all if it is numbered sequentially from 1 to 18 rather than having it go from 1-6, then 1-12.

Markdown doesn't allow for starting a numbered list from something other than 1. The intent is for the tabbed writing to show up under number 6, but hopefully the context is clear enough that it is read as describing number 7 - 18. 

Maybe it's way too much of a hack and it should just be removed. Or perhaps there's a better way?

Reply:
No that's fine